src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -1870,6 +1889,37 @@ protected override void ProcessRecord()
                         }
                     }
 
+                    // Handle the '-SecurityDescriptor' parameter
+
+                    IntPtr psecurityDescriptor = IntPtr.Zero;
+
+                    bool returnStatus = NativeMethods.ConvertStringSecurityDescriptorToSecurityDescriptorW(
+                                                    securityDescriptor,
+                                                    NativeMethods.stringSDRevision,
+                                                    psecurityDescriptor,
+                                                    IntPtr.Zero);
```

Comment:
I have no experience in c# and whatever done so far is by referring the docs available. But now I'm stuck and need help here to use the value from `psecurityDescriptor` to pass to `SetServiceObjectSecurity` method as a `DWORD`.

Reply:
In this case, perhaps you should use https://docs.microsoft.com/en-us/dotnet/api/system.security.accesscontrol.rawsecuritydescriptor.-ctor?redirectedfrom=MSDN&view=netcore-2.1#System_Security_AccessControl_RawSecurityDescriptor__ctor_System_String_

Reply:
Perhaps this helps https://stackoverflow.com/questions/15771998/how-to-give-a-user-permission-to-start-and-stop-a-particular-service-using-c-sha

Reply:
Thanks @iSazonov , that gave me more clarity. Will setting only DACL suffice here ? how about SACL , Owner, Group ?

Reply:
I reviewed the example from my link above and found it being very informative. There is "psd" (security descriptor) which contains all you ask.

Reply:
Thanks, I am able to set the DACL. Currently finding way to use `SE_SECURITY_NAME` privilege for getting `ACCESS_SYSTEM_SECURITY` permission in the handle.

Reply:
DACL is the actual permission, SACL is for auditing.  You would want to support both if both are given.  But it would also be ok to just document that only DACL is supported for now.

Look online for enabling security privilege in C#.

Reply:
Thanks @SteveL-MSFT . I looked into few docs online and found `AdjustTokenPrivileges` , but having some difficulty in using it w.r.t few parameters. This is becoming a great learning for me !

Reply:
I have added change for only DACL now a suggested. Will open an issue for SACL support to track. I will update the Docs once review is done.

Reply:
```
sc.exe sdshow wuauserv

D:(A;;CCLCSWRPLORC;;;AU)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;BA)(A;;CCDCLCSWRPWPDTLOCRSDRCWDWO;;;SY)
```

Reply:
My question was on which service should I take to change the setting ? I tested it by creating a dummy service in my box.

Reply:
You could do the same here (feature tagged test).

Reply:
Sure. Seems like because of https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Management/Set-Service.Tests.ps1#L19 , tests can fail. I could raise a PR fix this first .